Tools/TestWebKitAPI/Tests/WebKitCocoa/NavigationSwipeTests.mm

```diff
@@ -95,3 +101,44 @@ - (void)startInteractiveTransition:(id <UIViewControllerContextTransitioning>)tr
 }
 
 #endif // PLATFORM(IOS_FAMILY)
+
+#if PLATFORM(MAC)
+
+TEST(NavigationSwipeTests, WindowRelativeBoundsForCustomSwipeViews)
+{
+    RetainPtr customSwipeView = adoptNS([[NSView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600)]);
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600)]);
+    [webView setAllowsBackForwardNavigationGestures:YES];
+    [webView addSubview:customSwipeView.get()];
+    [webView _setCustomSwipeViews:@[ customSwipeView.get() ]];
+    [webView _setCustomSwipeViewsObscuredContentInsets:NSEdgeInsetsMake(100, 200, 50, 50)];
+
+    __auto_type boundsForCustomSwipeView = [webView _windowRelativeBoundsForCustomSwipeViewsForTesting];
```

Comment:
Nit: I know you're moving old  code around, but maybe just `auto`? They are identical in clang